plumbing/transport/server/server.go

```diff
@@ -79,6 +80,7 @@ func (h *handler) NewReceivePackSession(s storer.Storer) (transport.ReceivePackS
 }
 
 type session struct {
+	version  int
```

Comment:
Let's create a type for version which defaults to `v1` and gets populated based on the protocol negotiation.